src/Shared/FileUtilities.cs

```diff
@@ -421,6 +421,72 @@ internal static string FixFilePath(string path)
         }
 
         /// <summary>
+        /// If on Unix, convert backslashes to slashes for strings that resemble paths.
+        /// The heuristic is if something resembles paths (contains slashes) check if the
+        /// first segment exists and is a directory.
+        /// Use a native shared method to massage file path. If the file is adjusted,
+        /// that qualifies is as a path.
+        /// </summary>
+        internal static string MaybeAdjustFilePath(string value)
+        {
+            if (NativeMethodsShared.IsWindows || string.IsNullOrWhiteSpace(value) ||
+                value.StartsWith("$(") || value.StartsWith("@(") || value.StartsWith("\\\\"))
+            {
+                return value;
+            }
+
+            // For Unix-like systems, we may want to convert backslashes to slashes
+            string newValue = Regex.Replace(value, @"[\\/]+", "/");
```

Comment:
First, I think regex would only create one (output) string. Perf did not seem to be an issue. I ran a 1 min build with and without the function (for the run without the function I hand-fixed projects to have valid paths). The time was pretty much the same (few seconds difference was likely a precision error, especially considering that the build that actually changed the paths was a few sec faster).